project1/server/server.cpp

```diff
@@ -10,105 +10,32 @@
 
 #include "agent.h"
 #include "file_handler/file_handler.h"
+#include "../thread_pool/thread_pool.h"
+#include "server_tasks/nport_task.h"
+#include "server_tasks/tport_task.h"
 
 namespace server {
-namespace {
 
-using file_handler::FileHandler;
+    void Server::FtpService(uint16_t nPort, uint16_t tPort) {
 
-/// Maximum queue size to use for listening on the server socket.
-constexpr uint8_t kMaxQueueSize = 5;
+        thread_pool::ThreadPool pool;
 
-/**
- * @brief Helper function that creates the address structure to use for the
- *   server.
- * @param port The port to listen on.
- * @return The address structure.
- */
-struct sockaddr_in MakeAddress(uint16_t port) {
-  struct sockaddr_in address {};
-  address.sin_family = AF_INET;
-  address.sin_addr.s_addr = INADDR_ANY;
-  address.sin_port = htons(port);
+        std::shared_ptr<server_tasks::CommandIDs> active_ids;
+        auto nPortTask = std::make_shared<server_tasks::NPortTask>();
+        auto tPortTask = std::make_shared<server_tasks::TPortTask>();
 
-  return address;
-}
+        // pass active command list to nPortTask and tPortTask
+        nPortTask->SetActiveCommandIDs(active_ids);
```

Comment:
Could these not just be constructor parameters?

Reply:
The only reason I didn't do that initially is because I hadn't seen a task with a constructor. For some reason this gave me the notion to avoid constructors with tasks, but I see now that's unwarrented.

Reply:
Yep, constructors in tasks are fine. (I'm surprised you hadn't come across one before, because I definitely use them in the `ThreadPool` tests.